src/Collection.php

```diff
@@ -127,6 +135,10 @@ public function __construct(Manager $manager, string $databaseName, string $coll
             throw new InvalidArgumentException('$collectionName is invalid: ' . $collectionName);
         }
 
+        if (isset($options['codec']) && ! $options['codec'] instanceof DocumentCodec) {
```

Comment:
Since a codec library might return something other than an `object` instance, this requires a `DocumentCodec`. While there technically are ways to support multiple document codecs in a library, this would require a specific library that only accepts `DocumentCodec` instances. For this first iteration, I decided to only allow a `DocumentCodec` instance, as a potential future `DocumentCodecLibrary` would always implement this interface, so this is forward compatible with such a change.